spec/models/box_spec.rb

```diff
@@ -17,273 +11,96 @@
   let(:follow_upper) { create(:user, :follow_upper) }
   let(:inventory_type_research_needed) { create(:inventory_type, requires_research: true) }
   let(:inventory_type_no_research_needed) { create(:inventory_type, requires_research: false) }
+  let(:box_request_with_reviewer) { create(:box_request, :has_reviewer) }
+  let(:reviewed_box_request) { create(:box_request, :review_complete) }
+  let(:box) { create(:box, :is_designed)  }
 
   describe "state transitions" do
     before(:each) do
       allow(AutoEmailHandler).to receive(:new)
     end
 
     it "has state reviewed after box_request is reviewed" do
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
+      box_request_with_reviewer.claim_review!
+      box_request_with_reviewer.complete_review!
+      box = box_request_with_reviewer.box
       expect(box.aasm_state).to eq("reviewed")
     end
 
     it "transitions from reviewed to design_in_progress" do
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
-      box.designed_by_id = designer.id;
-      box.save
+      box = create(:box, :reviewed)
       expect(box).to transition_from(:reviewed).to(:design_in_progress).on_event(:claim_design)
     end
 
     it "transitions from design_in_progress to designed" do
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
+      box = build(:box, :design_in_progress)
       allow(box).to receive(:send_research_solicitation_email!)
-      box.designed_by_id = designer.id;
-      box.save
       box.claim_design!
       box.check_has_box_items # make sure there are items
-      # make sure at least one item needs research
+      #add items that need research
       create(:box_item, box: box, inventory_type: inventory_type_research_needed)
       expect(box).to transition_from(:design_in_progress).to(:designed).on_event(:complete_design)
       expect(box).to have_received(:send_research_solicitation_email!)
     end
 
     it "transitions from design_in_progress to researched when research not needed" do
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
+      box = build(:box, :design_in_progress)
       allow(box).to receive(:send_assembly_solicitation_email!)
```

Comment:
The test fails when I delete allow(box).to receive(:send_assembly_solicitation_email!). Does it need to be added somewhere in the factory? If so, should I do that for all the emails that should be received?

Reply:
@RachelWyatt hmm... it _should_ pass and shouldn't need to be in a factory. the allow, afaik, means that it's confirming that the method exists on `box`, rather than that it was executed.